src/OrchardCore/OrchardCore.Modules/ModularTenantContainerMiddleware.cs

```diff
@@ -76,11 +78,27 @@ public class ModularTenantContainerMiddleware
                         finally
                         {                            
                             semaphore.Release();
-                            _semaphores.TryRemove(shellSetting.Name, out semaphore);
+                            _semaphores.TryRemove(shellSettings.Name, out semaphore);
                         }
                     }
                     
                     await _next.Invoke(httpContext);
+                    deferredTaskEngine = scope.ServiceProvider.GetService<IDeferredTaskEngine>();
```

Comment:
Create a boolean instead of keeping `deferredTaskEngine` across scopes. -> `hasPendingTasks`

Reply:
To be clear, keep this line, but introduce a boolean that will be use below.